src/main/java/net/sf/mzmine/util/ScanUtils.java

```diff
@@ -603,4 +612,118 @@ public static int findBestFragmentScan(RawDataFile dataFile, Range<Double> rtRan
     return dataPoints;
   }
 
+  public static Stream<Scan> streamAllFragmentScans(PeakListRow row) {
+    return Arrays.stream(row.getAllMS2Fragmentations());
+  }
+
+  /**
+   * Sorted list (best first) of all MS2 fragmentation scans with n signals >= noiseLevel in the
+   * specified or first massList, if none was specified
+   * 
+   * @param row
+   * @param massListName the name or null/empty to always use the first masslist
+   * @param noiseLevel
+   * @param minNumberOfSignals
+   * @param sort the sorting property (best first, index=0)
+   * @return
+   */
+  @Nonnull
+  public static List<Scan> listAllFragmentScans(PeakListRow row, @Nullable String massListName,
+      double noiseLevel, int minNumberOfSignals, ScanSortMode sort)
+      throws MissingMassListException {
+    List<Scan> scans = listAllFragmentScans(row, massListName, noiseLevel, minNumberOfSignals);
+    // first entry is the best scan
+    scans.sort(Collections.reverseOrder(new ScanSorter(massListName, noiseLevel, sort)));
+    return scans;
+  }
+
+  /**
+   * List of all MS2 fragmentation scans with n signals >= noiseLevel in the specified or first
+   * massList, if none was specified
+   * 
+   * @param row
+   * @param massListName the name or null/empty to always use the first masslist
+   * @param noiseLevel
+   * @param minNumberOfSignals
+   * @return
+   */
+  @Nonnull
+  public static List<Scan> listAllFragmentScans(PeakListRow row, @Nullable String massListName,
+      double noiseLevel, int minNumberOfSignals) throws MissingMassListException {
+    List<Scan> filtered = new ArrayList<Scan>();
+    Scan[] scans = row.getAllMS2Fragmentations();
+    for (Scan scan : scans) {
+      // find mass list: with name or first
+      final MassList massList = getMassListOrFirst(scan, massListName);
+      if (massList == null)
+        throw new MissingMassListException("", massListName);
+
+      // minimum number of signals >= noiseLevel
+      int signals = 0;
+      for (DataPoint dp : massList.getDataPoints())
+        if (dp.getIntensity() >= noiseLevel)
+          signals++;
+      if (signals >= minNumberOfSignals)
+        filtered.add(scan);
+    }
+    return filtered;
+  }
+
+  /**
+   * Get specific masslist or the first if no masslist name is specified
+   * 
+   * @param scan
+   * @param massListName
+   * @return null if no masslist with this name or if name was not specified and this scan has zero
+   *         masslists
+   * @throws MissingMassListException
+   */
+  public static MassList getMassListOrFirst(Scan scan, String massListName) {
+    final MassList massList;
+    if (massListName == null || massListName.length() == 0) {
```

Comment:
Just FYI, there is a convenient ````Strings.isNullOrEmpty()```` method in Google Guava

Reply:
I have moved this class to util.scans